### PR TITLE
[bugfix] adding file extensions to import statements

### PR DESCRIPTION
**Pull request by nnmrts**
This should resolve issue #4203.

I used the search and replace tool of vscode to do this, with these two regexes:
```regex
import (.*?) ('|")(.*?)(\.js)('|")

import $1 $2$3$5
```

```regex
import (.*?) ('|")(.*?)('|")

import $1 $2$3.js$4
```

### CODE REVIEW

**Comment by nnmrts** [comment]
did a mini fix https://github.com/moment/moment/pull/4533/commits/1a8f348a1793c6717d0ebec9169fabc4e8cb32f2 because jscs threw an error

File: tasks/transpile.js [hunk 1]
```diff
@@ -155,11 +155,11 @@ module.exports = function (grunt) {
     function generateLocales(target, localeFiles, opts) {
         var files = localeFiles,
             code = [
-                'import moment from "./moment";',
+                'import moment from "./moment.js";',
                 'export default moment;'
             ].concat(files.map(function (file) {
                 var identifier = path.basename(file, '.js').replace('-', '_');
-                return 'import ' + identifier + ' from "./' + file + '";';
+                return 'import ' + identifier + ' from .js"./' + file + '";';
```
Comment (s7726) [inline tasks/transpile.js]: The addition of the `.js` in this line appears to be the cause of the failing tests.

Comment (nnmrts) [reply tasks/transpile.js]: yeah, i know...already fixed that locally, but there are still bugs...

**Comment by marwahaha** [comment]
This has been sitting for a long time, so I will close it.

Thanks for opening this request.
For now, we will not take this change.

**Pull request closed without merging**